xcodeproj/internal/bazel_integration_files/calculate_output_groups.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/python3
+
+import glob
+import json
+import os
+import sys
+import time
+import traceback
+
+def _calculate_build_request_file(objroot):
+    build_description_cache = max(
+        glob.iglob(f"{objroot}/XCBuildData/BuildDescriptionCacheIndex-*"),
+        key = os.path.getctime,
+    )
+    with open(build_description_cache, 'rb') as f:
+        f.seek(-32, os.SEEK_END)
+        build_request_id = f.read().decode('ASCII')
+    return f"{objroot}/XCBuildData/{build_request_id}-buildRequest.json"
+
+def _calculate_output_group_target_ids(
+        build_request_file,
+        scheme_target_ids,
+        guid_target_ids):
+    try:
+        # The first time a certain buildRequest is used, the buildRequest.json
+        # might not exist yet, so we wait a bit for it to exist
+        wait_counter = 0
+        time_to_wait = 10
+        while not os.path.exists(build_request_file):
+            time.sleep(1)
+            wait_counter += 1
+            if wait_counter > time_to_wait:
+                break
+
+        with open(build_request_file, encoding = "utf-8") as f:
+            # Parse the build-request.json file
+            build_request = json.load(f)
+
+        buildable_target_ids = []
+        for target in build_request["configuredTargets"]:
+            buildable_target_ids.extend(guid_target_ids[target["guid"]])
```

Comment:
These buildRequest and PIFCache files probably shouldn't be depended on. But the same logic can be used to say we should use XCBBuildServiceProxyKit. Since we safely handle API change, or the files just not existing anymore, this is "safe" to rely on, and we can come up with another solution in the future if it ever breaks.